src/ngResource/resource.js

```diff
@@ -702,12 +702,13 @@ angular.module('ngResource', ['ng']).
               return $q.reject(response);
             });
 
-            promise.finally(function() {
+            promise = promise.finally(function(response) {
```

Comment:
Fact: If a `resource` call is canceled the internal promise is rejected

The previous code did the following:

`internalPromise.finally(...)` this creates a promise that I will call `X`
`internalPromise.then(...)` this second promise is returned. I will call this promise `Y`

If `internalPromise` is rejected, then it will reject `X` and `Y`. The user can handle the rejection of `Y` as this is what they receive, but they can do nothing about `X`. This leaves us with 2 options

1/ Change the code to
```js
promise.finally(...).then(noop, noop);
promise = promise.then(...);
```

This will handle any rejections from `X`

2/ What is in this patch

```js
promise = promise.finally(...).then(...);
```

The second option has the benefit of not creating another promise